pallets/gear-builtin/src/lib.rs

```diff
@@ -158,7 +163,13 @@ pub mod pallet {
     pub struct Pallet<T>(PhantomData<T>);
 
     #[pallet::config]
-    pub trait Config: frame_system::Config {
+    pub trait Config: frame_system::Config + pallet_staking::Config {
```

Comment:
can we somehow avoid depending on staking pallet explicitly? if yes, place todo here pls

Reply:
It certainly can be done, but the only way I see right now is to explicitly declare the `Config::RuntimeCall` type being `From<gbuiltin_staking::Request>` and then defining the conversion at the runtime level (where the main `RuntimeCall` enum is defined in `vara/runtime/src/lib.rs`). But that would be much uglier choice and would require exposing internal staking builtin logic at the higher runtime level.

In my opinion declaring the pallet's `Config` trait implementing also `pallet_staking::Config` to be able to convert the `pallet_staking::Call<T>` type to the builtin pallet's `RuntimeCall` is not a problem at all - this is Substrate-idiomatic approach and should not be avoided.

Reply:
**Update:** removed the dependency.